akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/ActorThreadSpec.scala

```diff
@@ -0,0 +1,298 @@
+    }
+
+    "allow access from ask response mapper" in {
+      val probe = createTestProbe[String]()
+      val echo = spawn(Echo())
+
+      spawn(Behaviors.setup[String] { context =>
+        context.ask[Echo.Msg, Int](echo, Echo.Msg(18, _)) {
+          case Success(i) =>
+            // this is allowed because the mapping function is running in the target actor
+            context.children
+            i.toString
+          case Failure(e) => throw e
+        }
+
+        Behaviors.receiveMessage { msg =>
+          probe.ref ! msg
+          Behaviors.same
+        }
+      })
+
+      probe.expectMessage("18")
+    }
+
+    "detect wrong context in construction of AbstractBehavior" in {
+      val probe = createTestProbe[String]()
+      val ref = spawn(Behaviors.setup[String] { context =>
+        // missing setup new AbstractBehavior and passing in parent's context
+        val child = context.spawnAnonymous(new AbstractBehavior[String](context) {
+          override def onMessage(msg: String): Behavior[String] = {
+            probe.ref ! msg
+            Behaviors.same
+          }
+        })
+
+        Behaviors.receiveMessage { msg =>
+          child ! msg
+          Behaviors.same
+        }
+      })
+
+      // 2 occurrences because one from PostStop also
+      LoggingTestKit
+        .error[IllegalStateException]
+        .withMessageContains("was created with wrong ActorContext")
+        .withOccurrences(2)
+        .expect {
+          // it's not detected when spawned, but when processing message
+          ref ! "hello"
+          probe.expectNoMessage()
+        }
+    }
+
+    "detect illegal access from AbstractBehavior constructor" in {
+      val probe = createTestProbe[UnsupportedOperationException]()
+
+      spawn(Behaviors.setup[String] { context =>
+        context.spawnAnonymous(
+          Behaviors.setup[String](_ =>
+            // wrongly using parent's context
+            new AbstractBehavior[String](context) {
+              try {
+                this.context.children
+              } catch {
+                case e: UnsupportedOperationException =>
+                  probe.ref ! e
+              }
+
+              override def onMessage(msg: String): Behavior[String] = {
+                Behaviors.same
+              }
+            }))
+
+        Behaviors.empty
+      })
+
+      probe.receiveMessage().getMessage should include("[children]")
+    }
+
+    "detect illegal access from AbstractBehavior constructor when missing setup" in {
+      //FIXME
+      pending
+      val probe = createTestProbe[UnsupportedOperationException]()
+
+      spawn(Behaviors.setup[String] { context =>
+        // missing setup for the new AbstractBehavior and passing in parent's context
+        context.spawnAnonymous(new AbstractBehavior[String](context) {
+          try {
+            // FIXME this isn't detected as wrong because constructor is running in parent thread here (missing setup)
```

Comment:
can we fix this?

Reply:
I don't think we can detect that. It will be detected when the child receives the first message, though. That is tested two tests up in "detect wrong context in construction of AbstractBehavior". I'll remove this test.